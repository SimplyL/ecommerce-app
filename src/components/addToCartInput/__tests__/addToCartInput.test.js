import React from 'react';
import { shallow } from 'enzyme';

import { AddToCartInput } from '../index';

describe('AddToCartInput component', () => {
  let wrapper;
  const mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<AddToCartInput id="1" addProduct={mockFn} />);
  });

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call add function with correct parameters', () => {
    wrapper.find('AddButton').simulate('click');
    expect(mockFn).toBeCalledWith({ id: "1", quantity: 1 });
  });

  it('should change item quantity and call add function with correct quantity', () => {
    const event = { target: { value: 4 } };
    wrapper.find('AmountInput').simulate('change', event);
    wrapper.find('AddButton').simulate('click');
    expect(mockFn).toBeCalledWith({ id: "1", quantity: 4 });
  });
});
