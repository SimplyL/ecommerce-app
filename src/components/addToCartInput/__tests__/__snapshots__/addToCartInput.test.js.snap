// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCartInput component should render without crashing 1`] = `
<InputContainer>
  <AmountInput
    onChange={[Function]}
    type="number"
    value={1}
  />
  <AddButton
    onClick={[Function]}
  >
    Add to cart
  </AddButton>
</InputContainer>
`;
